### ISSUE TITLE
Ractive 0.8.1 breaking change with adaptors

### ISSUE THREAD

**Issue by ADantes** [ractivejs/ractive#2714]
Anytime a Backbone collection is updated, Ractive (using ractive-adaptors-backbone) is now replacing its reference with a plain array.  This affects both 0.8.0 and 0.8.1.

This code runs without error against 0.7.3, but the second console.assert fails in 0.8.x:

```
var c = new Backbone.Collection();
var v = new Ractive({
    adapt: ['Backbone'],
    data: function() {
        return {
                c: c
        };
    }
});

console.assert(v.get("c") === c); // View has a stored reference to the collection
c.reset([
    {item: 1},
    {item: 2}
]);
console.assert(v.get("c") === c); // This errors out.  v.get("c") is now returning a plain array of Backbone models
```

0.7.3: https://jsfiddle.net/pg2sw3zb/
0.8.1: https://jsfiddle.net/mq96w409/1/

I'm unclear whether this should be considered a bug with Ractive or the adaptor, so apologies if this is with the wrong project.

**Comment by paulocoghi** [comment]
@rstacruz could help. The adaptor needs a few fixes to stay compatible with recent versions of Ractive.

**Comment by evs-chris** [comment]
It looks like this is caused by the Ractive side of the adaptor storing the _real_ value on the adaptor as the `value` property. The adaptor is also using the `value` property to store a reference to the model :expressionless:. I think it's probably best/easiest to update the adaptor for now.

**Comment by evs-chris** [comment]
Well, I fixed part of it from Ractive. There is a sort-of breaking change in the way the ractive handles the set from the adaptor. It looks like it really should be calling update instead of set. I think it may be a result of an inconsistency in the adaptor API prior to 0.8, but I'm not sure.

When the adaptor gets an update from the backbone side, it should just tell ractive that it updated, but instead it calls `set` with the array that the collection holds. Ractive gets the array and sees that it no longer needs the wrapper, which no longer matches since an array is not a Collection, and tears it down. So, if you change the `ractive.set(keypath, collection.models);` to `ractive.update(keypath);` it looks like it starts working as expected.

**Comment by ADantes** [comment]
I'm still going through and testing it, but the above suggestion appears to fix the issues for 0.8.2 and is backwards compatible with 0.7.3.

**Comment by evs-chris** [comment]
I'm going to pass this off to #2512
